core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -30,6 +31,17 @@ public class SnapshotManager implements ManageSnapshots {
         ops.current() != null, "Cannot manage snapshots: table %s does not exist", tableName);
     this.transaction =
         new BaseTransaction(tableName, ops, BaseTransaction.TransactionType.SIMPLE, ops.refresh());
+    this.isTransactionExternal = false;
+  }
+
+  SnapshotManager(BaseTransaction transaction) {
+    Preconditions.checkNotNull(transaction, "Input transaction cannot be null");
+    Preconditions.checkNotNull(
+        transaction.underlyingOps().current(),
+        "Cannot manage snapshots: table %s does not exist",
+        transaction.tableName());
```

Comment:
Could we add tests for these 2 unhappy cases?

Reply:
sure, done